driver/src/main.rs

```diff
@@ -240,33 +240,32 @@ fn main() {
         web3,
     );
 
-    // NOTE: Keep the shadowed orderbook around so it doesn't get dropped and we
-    //   can pass a reference to the filtered orderbook reader.
-    let unfiltered_orderbook: Box<dyn StableXOrderBookReading + Sync> =
-        if options.use_shadowed_orderbook {
-            let shadow_orderbook = OnchainFilteredOrderBookReader::new(
-                contract.clone(),
-                options.auction_data_page_size,
-                &options.orderbook_filter,
-            );
-            let shadowed_orderbook =
-                ShadowedOrderbookReader::new(primary_orderbook.as_ref(), shadow_orderbook);
-            Box::new(shadowed_orderbook)
-        } else {
-            primary_orderbook
-        };
-
     info!("Orderbook filter: {:?}", options.orderbook_filter);
     let filtered_orderbook =
-        FilteredOrderbookReader::new(&*unfiltered_orderbook, options.orderbook_filter);
+        FilteredOrderbookReader::new(&*primary_orderbook, options.orderbook_filter.clone());
+
+    // NOTE: Keep the shadowed orderbook around so it doesn't get dropped and we
+    //   can pass a reference to the filtered orderbook reader.
+    let orderbook: Box<dyn StableXOrderBookReading + Sync> = if options.use_shadowed_orderbook {
+        let shadow_orderbook = OnchainFilteredOrderBookReader::new(
```

Comment:
If we start filtering accounts as well, we might get some more inconsistencies. Maybe we should additionally wrap the `shadow_orderbook` in a `FilteredOrderbookReader` as well.

Reply:
Agree however this change would have been a bit more involved (cf. https://github.com/gnosis/dex-services/issues/806)

Reply:
Oh good point!